gui/src/app/init/analytics.js

```diff
@@ -1,75 +1,88 @@
-/* Copyright (C) 2017 Canonical Ltd. */
+/* Copyright (C) 2019 Canonical Ltd. */
 'use strict';
 
-
-/**
-  Factory that returns a function for sending analytics to GA.
-
-  @param {String} controllerAPI For checking status of the user.
-  @param {String} dataLayer The GA dataLayer instance.
-  @return {function} The curryed function for sending analaytics.
-*/
-const sendAnalyticsFactory = function(controllerAPI, dataLayer) {
+class Analytics {
+  /**
+    Set up the analytics instance.
+    @param {String} dataLayer - The Google Tag Manager dataLayer instance.
+    @param {Object} data - Optional parameters:
+      - categories: an array of categories;
+      - getLabel: a function that will be used to generate the label string.
+  */
+  constructor(dataLayer, data={}) {
+    this.dataLayer = dataLayer;
+    this.categories = data.categories || [];
+    this.getLabel = data.getLabel;
```

Comment:
`getLabel` is a bit weird....you can only really set it on the initial instance at the app level. I could see some use case where you may want to have multiple `getLabel` calls as you traverse down the tree.

Maybe if we want to leave it as-is for now it should be renamed like 'globalLabels' or something. Something to indicate that it'll be prepended to every event.